bears/jinja2/Jinja2Bear.py

```diff
@@ -0,0 +1,370 @@
+        :param control_spacing:
+            The number of spaces required on each side of a control tag.
+        """
+        for m in self.CONTROL_END_REGEX.finditer(line):
+            label = m.group("label")
+            try:
+                expected_label, start_in_line = self.control_stack.pop()
+            except IndexError:
+                # an end tag without an opening tag
+                yield Result.from_values(
+                    origin=self,
+                    message="Control end tag has no corresponding start",
+                    file=filename,
+                    line=line_number,
+                    column=m.start(0) + 1,
+                    end_line=line_number + 1,
+                    end_column=m.end(0) + 1,
+                    severity=RESULT_SEVERITY.MAJOR)
+                # Continue to next line as we've detected a syntax error
+                # and label checking can give false results
+                continue
+
+            # check for spacing issues with this tag
+            if not has_required_spacing(m.group("content"), control_spacing):
+                yield self.handle_control_spacing_issue(
+                    file, filename, line, line_number, control_spacing, m)
+
+            # yield results for incorrect or missing end labels
+            if label is None and line_number != start_in_line:
+                diff = generate_label_diff(
+                    file, filename, line, line_number, m, expected_label)
+                yield Result.from_values(
+                    origin=self,
+                    message="Unlabeled control end tag",
+                    file=filename,
+                    line=line_number,
+                    column=m.start(0) + 1,
+                    end_line=line_number,
+                    end_column=m.end(0) + 1,
+                    diffs=diff)
+            elif label != expected_label and line_number != start_in_line:
+                diff = generate_label_diff(
+                    file, filename, line, line_number, m, expected_label)
+                yield Result.from_values(
+                    origin=self,
+                    message="End tag label does not match expected label",
+                    file=filename,
+                    line=line_number,
+                    column=m.start("label") + 1,
+                    end_line=line_number,
+                    end_column=m.end("label") + 1,
+                    diffs=diff)
+
+    def check_for_empty_control_stack(self, filename):
+        """
+        Checks if the control stack is empty and yields a Result for each
+        item left on the stack.
+
+        :param filename: The name of the file currently being inspected.
+        """
+        if len(self.control_stack) > 0:
+            for (_, line_number) in self.control_stack:
+                yield Result.from_values(
+                    origin=self,
+                    message="Control start tag has no corresponding end",
```

Comment:
As discussed [here](https://github.com/coala/coala-bears/pull/969#discussion_r86066166), I do have the expected label, which is not exactly equal to the tag and the tag in question is highlighted by the Result anyway.